qualtran/bloqs/chemistry/quad_fermion/givens_bloq.py

```diff
@@ -0,0 +1,218 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+r"""The Givens rotation Bloqs help count costs for similarity transforming
+fermionic ladder operators to produce linear combinations of fermionic ladder operators.
+
+Following notation from Reference [1] we note that a single 
+ladder operator can be similarity transformed by a basis rotation to produce a linear 
+combination of ladder operators
+$$
+U(Q)a_{q}U(Q)^{\dagger} = \sum_{p}Q_{pq}^{*}a_{p} = \overrightarrow{a}_{q}\\
+U(Q)a_{q}^{\dagger}U(Q)^{\dagger} = \sum_{p}Q_{pq}a_{p}^{\dagger} = \overrightarrow{a}_{q}^{\dagger}
+$$
+Each vector of operators can be implemented by a $N$ (size of basis) Givens rotation unitaries as
+$$
+V_{\overrightarrow{Q}_{q}} a_{0} V_{\overrightarrow{Q}_{q}}^{\dagger} = \overrightarrow{a}_{q} \\
+V_{\overrightarrow{Q}_{q}} a_{0}^{\dagger} V_{\overrightarrow{Q}_{q}}^{\dagger} = \overrightarrow{a}_{q}^{\dagger}
+$$
+where 
+$$
+V_{\overrightarrow{Q}_{q}} = V_{n-1,n-2}(0, \phi_{n-1}) V_{n-2, n-3}(\theta_{n-2}, \phi_{n-2})V_{n-3,n-4}(\theta_{n-2}, \phi_{n-2})...V_{2, 1}(\theta_{1}, \phi_{1})V_{1, 0}(\theta_{0}, \phi_{0})
```

Comment:
nit: We tend to enforce a 100 character line limit for any docstrings (excluding URL references in the Refenreces section. I think here and in some other places you're over that.